core/templates/components/common-layout-directives/common-elements/confirm-or-cancel-modal.controller.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for simple modal with only two actions: close or
+ * dismiss.
+ */
+
+angular.module('oppia').controller('ConfirmOrCancelModalController', [
+  '$scope', '$uibModalInstance',
+  function($scope, $uibModalInstance) {
+    $scope.close = function(message) {
+      $uibModalInstance.close(message);
+    };
```

Comment:
Shouldn't this be `$scope.confirm = `, `close` and `cancel` seem quite the same.

Reply:
They actually are the same thing. The only difference is close will make the promise (from result property returned from $uibModal.open) fulfilled and the cancel (dismiss) will make the promise rejected.

Reply:
Sure, but why name it `close` and not `confirm`, I think that `confirm` is better here since it implies some kind of success. Also, the controller is named **`Confirm`**`OrCancelModalController`.

Reply:
It makes totally sense, I've renamed it. PTAL!